draft-ietf-dnsop-svcb-https.md

```diff
@@ -759,19 +759,44 @@ empty.
 Each scheme that uses this SvcParamKey defines a
 "default set" of supported ALPNs, which SHOULD NOT
 be empty.  To determine the set of protocol suites supported by an
-endpoint (the "ALPN set"), the client parses the set of ALPN identifiers in
-the "alpn" parameter, and then adds the default set unless the
-"no-default-alpn" SvcParamKey is present.  The presence of a value in
-the alpn set indicates that this service endpoint, described by
+endpoint (the "SVCB ALPN set"), the client parses the `alpn-value` to produce
+a set of `alpn-id`s, and then adds the default set unless the
+"no-default-alpn" SvcParamKey is present.  The presence of an ALPN protocol in
+the SVCB ALPN set indicates that this service endpoint, described by
 SvcDomainName and the other parameters (e.g. "port") offers service with
-the protocol suite associated with the ALPN ID.
+the protocol suite associated with this ALPN protocol.
 
-ALPN IDs that do not uniquely identify a protocol suite (e.g. an ID that
+ALPN protocol names that do not uniquely identify a protocol suite (e.g. an
+Identification Sequence that
 can be used with both TLS and DTLS) are not compatible with this
-SvcParamKey and MUST NOT be included in the ALPN set.
-
-Clients SHOULD NOT attempt connection to a service endpoint whose
-ALPN set does not contain any compatible protocol suites.  To ensure
+SvcParamKey and MUST NOT be included in the SVCB ALPN set.
+
+To establish a connection to the endpoint, clients MUST
+
+1. Let SVCB-ALPN-Intersection be the set of protocols in the SVCB ALPN set
+that the client supports.
+2. Let Intersection-Transports be the set of transports (e.g. TLS, DTLS, QUIC)
+implied by the protocols in SVCB-ALPN-Intersection.
+3. For each transport in Intersection-Transports, construct a ProtocolNameList
+containing the Identification Sequences of all the client's supported ALPN
+protocols for that transport, without regard to the SVCB ALPN set.
```

Comment:
FWIW, these steps are exactly what I'm implementing in our client already, so this seems to be the most practical approach.